src/Microsoft.Health.Fhir.Api/Features/Routing/UrlResolver.cs

```diff
@@ -113,13 +113,13 @@ public Uri ResolveResourceUrl(Resource resource, bool includeVersion = false)
             return new Uri(uriString);
         }
 
-        public Uri ResolveRouteUrl(IEnumerable<Tuple<string, string>> unsupportedSearchParams = null, string continuationToken = null)
+        public Uri ResolveRouteUrl(IEnumerable<Tuple<string, string>> unsupportedSearchParams = null, string continuationToken = null, IDictionary<string, object> routeValues = null)
```

Comment:
Is this required since we already have access to `Request.Query`. I think you just need to ensure that `_before` doesn't end up in `unsupportedSearchParams`. //cc @jackliums

Reply:
When a caller does not specify a value for _before in the query, we will pick a specific time to bound the query, and then add it to the selflink so that caller can know what exact timestamp was used.  This timestamp can be used for a new query to get a new batch of history records.  I am open to thoughts/suggestions on a better approach.

---
In reply to: [282159094](https://github.com/microsoft/fhir-server/pull/455#discussion_r282159094) [](ancestors = 282159094)

Reply:
I have reverted the code that added the _before param when none was specified by the caller.